### PR TITLE
Segmentation Survey: Fix saving answers error when leaving the segmentation survey step

### LINKED ISSUE #89851: Skip question when submitting empty answers

### LINKED ISSUE #90040: Move survey_key to a component prop

### LINKED ISSUE #90041: Fix context lost when saving answer on the last page

### LINKED ISSUE #90042: Remove feature flag

### PR DESCRIPTION
**Pull request by phcp**
Closes https://github.com/Automattic/wp-calypso/issues/89851
Closes https://github.com/Automattic/wp-calypso/issues/90040
Closes https://github.com/Automattic/wp-calypso/issues/90041
Closes https://github.com/Automattic/wp-calypso/issues/90042

## Proposed Changes

* Extracted the `SegmentationSurvey` component and navigation functions
  * Removed the dependency from the `SurveyContext` to make it easier to use
  * Moved functions from the provider to a hook, also removing the NavigationControls dependency
  * Added a surveyKey prop to the main component
* Fixed the saving issue due to the early routing when moving to the next step
* Disabled clickable items when the answers are being saved
* Sent 'skip' keyword as an answer key for skipped or empty questions
* Removed feature flag
* Logged save mutation errors

## Testing Instructions

* Apply this PR to your local
* Go to http://calypso.localhost:3000/setup/entrepreneur/start
* Verify if the survey was rendered as expected
* Open the Network tab on Dev Tools to verify if the questions are being saved
* Answer the first question and click Continue
* All the clickable items should be disabled while the request is in progress
* Submit the second question and verify if the answer was saved
* Reopen http://calypso.localhost:3000/setup/entrepreneur/start
  * The survey should be rendered without preloaded answers (localStorage cleaned)
* Perform regression tests on tracks events
  * Check if failed requests are correctly tracked
* Perform regression tests on saving skipped questions

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/segmentation-survey/index.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	const onContinue = useCallback(
+		async ( currentQuestion: Question ) => {
+			await handleSave( currentQuestion, answers[ currentQuestion.key ] || [] );
+		},
+		[ answers, handleSave ]
+	);
+
+	const onSkip = useCallback(
+		async ( currentQuestion: Question ) => {
+			await handleSave( currentQuestion, [ 'skipp' ] );
```
Comment (ivan-ottinger) [inline client/components/segmentation-survey/index.tsx]: Small typo, but will result in skipping not working:

Markup on 2024-04-30 at 17:30:33

```suggestion
			await handleSave( currentQuestion, [ 'skip' ] );
```

**Review by ivan-ottinger — COMMENTED** [review]
Nice! Many great improvements! 👍🏼👍🏼 

I haven't gotten to code review yet, but so far the survey works well in my tests.

I only noticed three things:

1. typo in the question skipping logic (more in the inline comment below)

2. Tracks cannot accept nested array (more in the inline comment below)

3. checkbox animation not working correctly (but this is likely unrelated to this PR and was probably introduced earlier)

Screen Capture on 2024-04-30 at 17-28-55

→ Looks like the check icon is appearing with delay instead of the animation we had earlier.

File: client/components/segmentation-survey/index.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+					clearAnswers();
+					onComplete?.();
+				}
+			} catch ( e ) {
+				const error = e as Error;
+
+				recordTracksEvent( 'calypso_segmentation_survey_error', {
+					survey_key: surveyKey,
+					question_key: currentQuestion.key,
+					answer_keys: answerKeys,
```
Comment (ivan-ottinger) [inline client/components/segmentation-survey/index.tsx]: Looks like Tracks won't allow us to pass a nested array here.

Markup on 2024-04-30 at 17:49:59

Instead we could consider something like:

```suggestion
					answer_keys: JSON.stringify( answerKeys ),
```

File: client/components/segmentation-survey/index.tsx (hunk 1 above)
Comment (phcp) [reply client/components/segmentation-survey/index.tsx]: Indeed, I forgot to remove it, it was only for testing the error logs. Thank you for catching that Ivan!

File: client/components/segmentation-survey/index.tsx (hunk 2 above)
Comment (phcp) [reply client/components/segmentation-survey/index.tsx]: Nice catch! I changed to `.join( ',' )` to match the other events format.

**Comment by phcp** [comment]
> → Looks like the check icon is appearing with delay instead of the animation we had earlier

Interesting, I can't reproduce it here, on my env, it works correctly. Maybe we can look at it on Thursday at your end @ivan-ottinger.

File: client/components/segmentation-survey/index.tsx (hunk 2 above)
Comment (ivan-ottinger) [reply client/components/segmentation-survey/index.tsx]: > I changed to .join( ',' ) to match the other events format.

That's much better than my initial proposal. 🙂

**Comment by ivan-ottinger** [comment]
> > → Looks like the check icon is appearing with delay instead of the animation we had earlier
> 
> Interesting, I can't reproduce it here, on my env, it works correctly. Maybe we can look at it on Thursday at your end @ivan-ottinger.

Yeah, looks like something in my environment. Doesn't seem to be browser related, as I was able to reproduce in both Chrome and Firefox.

Yes, a quick call could be nice - even to discuss some of the changes generally! 🙂

**Comment by ivan-ottinger** [comment]
The proposed changes look great! I didn't notice any other issues.

I appreciate the context removal as it makes the whole code much more readable. it is also great for reusability that we can pass the `surveyKey` as a prop to the main component now.

**Review by ivan-ottinger — APPROVED** [review]

**Pull request merged**